### PR TITLE
draft t2 decryption

### PR DESCRIPTION
**Pull request by tomt1664**

### CODE REVIEW

**Comment by lawlawlaw** [comment]
Fails the tests. See below


  Finished test [unoptimized + debuginfo] target(s) in 1m 17s
     Running target/debug/deps/tests-f8e3e733b948882b

running 3 tests
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: ParseIntError { kind: InvalidDigit }', server/src/enclave.rs:1605:42
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: ParseIntError { kind: InvalidDigit }', server/src/enclave.rs:1605:42
test tests::test_keygen ... FAILED
test tests::test_sign_keygen ... FAILED
second_message: get sealed secrets
second_message: got sealed secrets
"s1"
Secp256k1Scalar { purpose: "from_big_int", fe: SecretKey(62048ec3d18ef68738d943881fc915b90602de6d2460b5bcd25ed00b148816c6) }
"t2enc"
"04c824f5c52d9411f83f2f1fd08e659c8fd815384e5eb6e0c63af0c6ffaccc49a181938f4fb92eacd25ddd374935a50d34c60181d05f577cf232fb03743f142a51d99a9702bb0a479d9c397b0811c905bfff764d37cff264f71d45cc3f7d537f0f71a8d14948be16a8d1a3508f575c06f743dc466a51a4674c0cefeba53535ac1e"
thread '<unnamed>' panicked at 'assertion failed: nonce.len() == 12', /root/.cargo/git/checkouts/rust-crypto-sgx-81e970543180aadf/4cdd9f9/src/aes_gcm.rs:25:9
fatal runtime error: failed to initiate panic, error 5
test tests::test_transfer_sign_keygen ... FAILED

failures:

---- tests::test_keygen stdout ----
thread 'tests::test_keygen' panicked at 'called `Result::unwrap()` on an `Err` value: Generic("Error derserialising POST response: : EOF while parsing a value at line 1 column 0")', src/lib.rs:230:120
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- tests::test_sign_keygen stdout ----
thread 'tests::test_sign_keygen' panicked at 'called `Result::unwrap()` on an `Err` value: Generic("Error derserialising POST response: : EOF while parsing a value at line 1 column 0")', src/lib.rs:281:120

---- tests::test_transfer_sign_keygen stdout ----
verify - cond1: true, cond2: true
keyupdate first
thread 'tests::test_transfer_sign_keygen' panicked at 'called `Result::unwrap()` on an `Err` value: Generic("error sending request for url (http://0.0.0.0:8000/ecdsa/keyupdate/first): connection closed before message completed")', src/lib.rs:503:74


failures:
    tests::test_keygen
    tests::test_sign_keygen
    tests::test_transfer_sign_keygen

test result: FAILED. 0 passed; 3 failed; 0 ignored; 0 measured; 0 filtered out

error: test failed, to rerun pass '--lib'
The command '/bin/sh -c set -x     && mkdir $LOCKBOX_DB_PATH     && apt update     && apt install -y libgmp-dev llvm clang     && git clone -b master --single-branch https://github.com/apache/incubator-teaclave-sgx-sdk.git /root/sgx     && cd /opt/intel     && wget https://download.01.org/intel-sgx/sgx-linux/2.11/distro/ubuntu18.04-server/sgx_linux_x64_sdk_2.11.100.2.bin     && chmod +x sgx_linux_x64_sdk_2.11.100.2.bin     && echo yes | ./sgx_linux_x64_sdk_2.11.100.2.bin     && rm -f sgx_linux_x64_sdk_2.11.100.2.bin     && export SGX_SDK=/opt/intel/sgxsdk     && export PATH=$PATH:/root/.cargo/bin:$SGX_SDK/bin/x64:$SGX_SDK/bin     && export PKG_CONFIG_PATH=$SGX_SDK/pkgconfig     && export BINUTILS_PREFIX=/usr     && cd /root/lockbox     && echo "$TESTS"     && if [ "$TESTS" = "true" ] ; then sed -i 's/SGX_MODE ?= HW/SGX_MODE ?= SW/g' Makefile     && bash -c "source /opt/intel/sgxsdk/environment && SGX_MODE=SW make" && /docker-entrypoint.sh tests     && cd integration-tests && cargo test --no-default-features -- --test-threads=4 ; else make ; fi     && rm -rf /var/lib/apt/lists/*' returned a non-zero code: 101

File: integration-tests/src/lib.rs [hunk 1]
```diff
@@ -172,35 +174,11 @@ where
...
     match serde_json::from_str(value.as_str()) {
-	Ok(r) => Ok(r),
-	Err(e) => {
-	    Err(CError::Generic(format!("Error derserialising POST response: {}: {}", value.as_str(), e)))
-	}
+    Ok(r) => Ok(r),
+    Err(e) => {
+        Err(CError::Generic(format!("Error derserialising POST response: {}: {}", value.as_str(), e)))
     }
-}
```
Comment (lawlawlaw) [inline integration-tests/src/lib.rs:180]: Why delete this?

File: integration-tests/src/lib.rs [hunk 2]
```diff
@@ -232,62 +212,10 @@ pub fn verify(r: &BigInt, s: &BigInt, pubkey: &GE, message: &BigInt) -> bool {
 mod tests {
     use super::*;
 
-    fn init_dh() -> Lockbox {
```
Comment (lawlawlaw) [inline integration-tests/src/lib.rs:235]: Why delete this?

**Review by lawlawlaw — CHANGES_REQUESTED** [review]
Tests fail (see comment on PR page). Query deletions (see comments).

File: integration-tests/src/lib.rs (hunk 1 above)
Comment (tomt1664) [reply integration-tests/src/lib.rs:180]: Old version of the tests. Reverted.

File: integration-tests/src/lib.rs (hunk 2 above)
Comment (tomt1664) [reply integration-tests/src/lib.rs:235]: Old version of the test. Reverted.

**Comment by tomt1664** [comment]
The forked ECIES for the enclave now uses branch `sgx2` (i.e. ecies = { git = "https://github.com/commerceblock/rs.git", branch = "sgx2" }). All tests pass for this branch. 

This branch has the OpenSSL default removed and has dependencies for `RustCrypto` for the AES_256_GCM and the older `rust-crypto` for the `hkdf` function (which has a different API). 

All integration tests should now pass.

**Comment by tomt1664** [comment]
Merge now to master for testing - can always revert.

**Pull request merged**